pontoon/sync/tests/formats/test_compare_locales.py

```diff
@@ -0,0 +1,254 @@
+
+    def test_parse_empty_translation(self):
+        self.run_parse_empty_translation(BASE_ANDROID_XML_FILE, 3)
+
+    def test_save_basic(self):
+        input_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Comment -->
+    <string name="Source String">Source String</string>
+</resources>
+        """)
+        expected_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Comment -->
+    <string name="Source String">New Translated String</string>
+</resources>
+        """)
+
+        self.run_save_basic(input_string, expected_string, source_string=input_string)
+
+    def test_save_remove(self):
+        """Deleting strings removes them completely from the DTD file."""
+        input_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Comment -->
+    <string name="Source String">Source String</string>
+</resources>
+        """)
+        expected_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    </resources>
+        """)
+
+        self.run_save_remove(input_string, expected_string, source_string=input_string)
+
+    def test_save_source_removed(self):
+        """
+        If an entity is missing from the source resource, remove it from
+        the translated resource.
+        """
+        source_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="Source String">Source String</string>
+</resources>
+        """)
+        input_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="Missing Source String">Translated Missing String</string>
+    <string name="Source String">Translated String</string>
+</resources>
+        """)
+        expected_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="Source String">Translated String</string>
+</resources>
+        """)
+
+        self.run_save_no_changes(input_string, expected_string, source_string=source_string)
+
+    def test_save_source_no_translation(self):
+        """
+        If an entity is missing from the translated resource and has no
+        translation, do not add it back in.
+        """
+        source_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="Source String">Source String</string>
+    <string name="Other Source String">Other String</string>
+</resources>
+        """)
+        input_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="Other Source String">Translated Other String</string>
+</resources>
+        """)
+
+        self.run_save_no_changes(input_string, input_string, source_string=source_string)
+
+    def test_save_translation_missing(self):
+        source_string = dedent("""<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="String">Source String</string>
+    <string name="Missing String">Missing Source String</string>
```

Comment:
Can you add a comment for "Missing String"? It should also show up in the serialized file, it'd be good to have a check for that here as well.